src/nodes/model/custom/Custom.ts

```diff
@@ -0,0 +1,45 @@
+import { Node } from '@baklavajs/core';
+import { Layers, Nodes } from '@/nodes/model/Types';
+
+export default class Custom extends Node {
+  type = Layers.Custom;
+  name = Nodes.Custom;
+
+  private numberOfInputs = 0;
+
+  constructor() {
+    super();
+    // TODO FE-38 Add minimum value of 0 to this option.
+    this.addOption('Number of Inputs', 'IntegerOption', 0);
+    this.addOutputInterface('Output');
+    this.events.update.addListener(this, (event: any) => {
+      this.nodeUpdated(event);
+    });
+  }
+
+  private nodeUpdated(event: any) {
+    if (event.name === 'Number of Inputs') {
+      if (event.option) {
+        if (event.option.value > this.numberOfInputs) {
+          this.addInputs(event.option.value - this.numberOfInputs);
+        } else if (event.option.value < this.numberOfInputs) {
+          this.removeInputs(this.numberOfInputs - event.option.value);
+        }
+      }
+    }
+  }
+
+  private addInputs(n: number) {
+    Array.from(Array(n).keys()).forEach((i: number) => {
+      this.addInputInterface(`Input ${this.numberOfInputs + i + 1}`);
+    });
+    this.numberOfInputs += n;
```

Comment:
Because I don't know TypeScript, could you just explain to me why this is better than a for loop? Or are they not are allowed or something. It just seems odd (but I don't know any better) that you're making an array in order to iterate over it...

Reply:
Yes, so I think `for..in` loops are not allowed because they iterate over the whole prototype of the object, and not just keys like in an Array. This seems to be the common way of simply iterating over a range of integers in Typescript so I just did this aha.

Reply:
Alrighty, I'll take your word for it haha